panic when storage.capacity is empty permanently disables the operator and webhooks until the resource is removed

**Describe the bug**
Omitting storage.capacity leads to controller panic and not being able to edit the resource because webhooks are down.

**To Reproduce**
Steps to reproduce the behavior:
1. Create ScyllaCluster without storage.capacity

**Expected behavior**
Validation should forbid it if it's required, but changing it now would break backwards compatibility. Fixing the panic is best we can do for v1.

The offending line is https://github.com/scylladb/scylla-operator/blob/dc3f32a7286b3fcc045e8e6f98070a4baa401bb4/pkg/controllers/cluster/resource/resource.go#L343

**Logs**
E0301 14:00:52.106176       1 runtime.go:78] Observed a panic: &errors.errorString{s:"cannot parse '': quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$'"} (cannot parse '': quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$')
goroutine 401 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x19c4a60, 0xc00005a500)
	k8s.io/apimachinery@v0.18.4/pkg/util/runtime/runtime.go:74 +0xa6
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	k8s.io/apimachinery@v0.18.4/pkg/util/runtime/runtime.go:48 +0x89
panic(0x19c4a60, 0xc00005a500)
	runtime/panic.go:969 +0x1b9
k8s.io/apimachinery/pkg/api/resource.MustParse(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	k8s.io/apimachinery@v0.18.4/pkg/api/resource/quantity.go:127 +0x1c8
github.com/scylladb/scylla-operator/pkg/controllers/cluster/resource.StatefulSetForRack(0xc00083a4c4, 0xa, 0x3, 0x0, 0x0, 0x0, 0x0, 0xc000010ab0, 0xc000010ae0, 0x0, ...)
	github.com/scylladb/scylla-operator/pkg/controllers/cluster/resource/resource.go:347 +0xb85
github.com/scylladb/scylla-operator/pkg/controllers/cluster/actions.(*RackCreate).Execute(0xc000b7c000, 0x1eebda0, 0xc000010b10, 0xc000010ff0, 0x1eebda0, 0xc000010b10)
	github.com/scylladb/scylla-operator/pkg/controllers/cluster/actions/create.go:42 +0xd0
github.com/scylladb/scylla-operator/pkg/controllers/cluster.(*ClusterReconciler).sync(0xc000020380, 0xc000655d40, 0x0, 0x0)
	github.com/scylladb/scylla-operator/pkg/controllers/cluster/sync.go:72 +0xaed
github.com/scylladb/scylla-operator/pkg/controllers/cluster.(*ClusterReconciler).Reconcile(0xc000020380, 0xc0005b9680, 0x22, 0xc000059834, 0x5, 0x87f5266a, 0xc000486510, 0xc0007f3728, 0xc0007f3710)
	github.com/scylladb/scylla-operator/pkg/controllers/cluster/cluster_controller.go:122 +0x7c9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000866120, 0x1ab8fa0, 0xc00064e8a0, 0xc000374900)
	sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:233 +0x166
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000866120, 0x203000)
	sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:209 +0xb0
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc000866120)
	sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:188 +0x2b
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000a76f40)
	k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000a76f40, 0x1eabe40, 0xc000a791d0, 0x1, 0xc0005ea060)
	k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000a76f40, 0x3b9aca00, 0x0, 0x1d4b401, 0xc0005ea060)
	k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc000a76f40, 0x3b9aca00, 0xc0005ea060)
	k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:90 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:170 +0x3fa
panic: cannot parse '': quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$' [recovered]
	panic: cannot parse '': quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$'

--- a/pkg/controllers/cluster/actions/create.go
+++ b/pkg/controllers/cluster/actions/create.go
@@ -39,9 +39,13 @@ func (a *RackCreate) Name() string {
 
 func (a *RackCreate) Execute(ctx context.Context, s *State) error {
 	r, c := a.Rack, a.Cluster
-	newSts := resource.StatefulSetForRack(r, c, a.OperatorImage)
+	newSts, err := resource.StatefulSetForRack(r, c, a.OperatorImage)
+	if err != nil {
+		return err
+	}
+
 	existingSts := &appsv1.StatefulSet{}
-	err := s.Get(ctx, naming.NamespacedNameForObject(newSts), existingSts)
+	err = s.Get(ctx, naming.NamespacedNameForObject(newSts), existingSts)
 	// Check if StatefulSet already exists
 	// TODO: Check this logic
 	if err == nil {

--- a/pkg/controllers/cluster/actions/rack_synchronized_test.go
+++ b/pkg/controllers/cluster/actions/rack_synchronized_test.go
@@ -48,7 +48,10 @@ func TestRackSynchronizedAction_SubActionUpdatesRack(t *testing.T) {
 	})
 
 	cluster := unit.NewMultiRackCluster(1)
-	firstRack := resource.StatefulSetForRack(cluster.Spec.Datacenter.Racks[0], cluster, "image")
+	firstRack, err := resource.StatefulSetForRack(cluster.Spec.Datacenter.Racks[0], cluster, "image")
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	objects := []runtime.Object{firstRack}
 	kubeClient := fake.NewSimpleClientset(objects...)
@@ -94,11 +97,18 @@ func TestRackSynchronizedAction_RackAreUpgradedInSequence(t *testing.T) {
 	})
 
 	cluster := unit.NewMultiRackCluster(1, 1)
-	firstRack := resource.StatefulSetForRack(cluster.Spec.Datacenter.Racks[0], cluster, "image")
+	firstRack, err := resource.StatefulSetForRack(cluster.Spec.Datacenter.Racks[0], cluster, "image")
+	if err != nil {
+		t.Fatal(err)
+	}
 	firstRackReady := firstRack.DeepCopy()
 	firstRackReady.Status.ObservedGeneration = firstRackReady.Generation
 	firstRackReady.Status.ReadyReplicas = cluster.Spec.Datacenter.Racks[0].Members
-	secondRack := resource.StatefulSetForRack(cluster.Spec.Datacenter.Racks[1], cluster, "image")
+
+	secondRack, err := resource.StatefulSetForRack(cluster.Spec.Datacenter.Racks[1], cluster, "image")
+	if err != nil {
+		t.Fatal(err)
+	}
 	secondRackReady := secondRack.DeepCopy()
 	secondRackReady.Status.ObservedGeneration = secondRackReady.Generation
 	secondRackReady.Status.ReadyReplicas = cluster.Spec.Datacenter.Racks[1].Members

--- a/pkg/controllers/cluster/actions/sidecar_upgrade_test.go
+++ b/pkg/controllers/cluster/actions/sidecar_upgrade_test.go
@@ -35,7 +35,10 @@ func TestSidecarUpgradeAction(t *testing.T) {
 
 	cluster := unit.NewMultiRackCluster(1)
 	rack := cluster.Spec.Datacenter.Racks[0]
-	rackSts := resource.StatefulSetForRack(rack, cluster, preUpdateImage)
+	rackSts, err := resource.StatefulSetForRack(rack, cluster, preUpdateImage)
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	updatedContainer := corev1.Container{
 		Name:    naming.SidecarInjectorContainerName,

--- a/pkg/controllers/cluster/actions/upgrade_version_test.go
+++ b/pkg/controllers/cluster/actions/upgrade_version_test.go
@@ -16,7 +16,10 @@ func TestUpgradeStatefulSetScyllaImage(t *testing.T) {
 	ctx := context.Background()
 	cluster := unit.NewSingleRackCluster(3)
 	rack := cluster.Spec.Datacenter.Racks[0]
-	sts := resource.StatefulSetForRack(rack, cluster, "sidecar")
+	sts, err := resource.StatefulSetForRack(rack, cluster, "sidecar")
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	tests := []struct {
 		name          string

--- a/pkg/controllers/cluster/resource/resource.go
+++ b/pkg/controllers/cluster/resource/resource.go
@@ -111,13 +111,19 @@ func memberServicePorts(cluster *scyllav1.ScyllaCluster) []corev1.ServicePort {
 	return ports
 }
 
-func StatefulSetForRack(r scyllav1.RackSpec, c *scyllav1.ScyllaCluster, sidecarImage string) *appsv1.StatefulSet {
+func StatefulSetForRack(r scyllav1.RackSpec, c *scyllav1.ScyllaCluster, sidecarImage string) (*appsv1.StatefulSet, error) {
 	rackLabels := naming.RackLabels(r, c)
 	placement := r.Placement
 	if placement == nil {
 		placement = &scyllav1.PlacementSpec{}
 	}
 	opt := true
+
+	storageCapacity, err := resource.ParseQuantity(r.Storage.Capacity)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse '%v': %v", r.Storage.Capacity, err)
+	}
+
 	sts := &appsv1.StatefulSet{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:            naming.StatefulSetNameForRack(r, c),
@@ -346,7 +352,7 @@ func StatefulSetForRack(r scyllav1.RackSpec, c *scyllav1.ScyllaCluster, sidecarI
 						StorageClassName: r.Storage.StorageClassName,
 						Resources: corev1.ResourceRequirements{
 							Requests: corev1.ResourceList{
-								corev1.ResourceStorage: resource.MustParse(r.Storage.Capacity),
+								corev1.ResourceStorage: storageCapacity,
 							},
 						},
 					},
@@ -368,7 +374,7 @@ func StatefulSetForRack(r scyllav1.RackSpec, c *scyllav1.ScyllaCluster, sidecarI
 			sts.Spec.Template.Spec.Volumes, *Volume.DeepCopy())
 	}
 	sts.Spec.Template.Spec.Containers = append(sts.Spec.Template.Spec.Containers, agentContainer(r, c))
-	return sts
+	return sts, nil
 }
 
 func containerPorts(c *scyllav1.ScyllaCluster) []corev1.ContainerPort {

--- a/pkg/controllers/cluster/status_test.go
+++ b/pkg/controllers/cluster/status_test.go
@@ -68,7 +68,10 @@ func TestClusterStatus_SourceOfRackStatus(t *testing.T) {
 		},
 	}
 
-	rackSts := resource.StatefulSetForRack(rack, cluster, "image")
+	rackSts, err := resource.StatefulSetForRack(rack, cluster, "image")
+	if err != nil {
+		t.Fatal(err)
+	}
 	rackSts.Spec.Replicas = pointer.Int32Ptr(123)
 	rackSts.Status = appsv1.StatefulSetStatus{
 		Replicas:      2,

--- a/pkg/controllers/cluster/sync.go
+++ b/pkg/controllers/cluster/sync.go
@@ -167,7 +167,11 @@ func (cc *ClusterReconciler) nextAction(ctx context.Context, cluster *scyllav1.S
 }
 
 func (cc *ClusterReconciler) sidecarUpdateNeeded(ctx context.Context, rack scyllav1.RackSpec, cluster *scyllav1.ScyllaCluster) (bool, corev1.Container, error) {
-	desiredSts := resource.StatefulSetForRack(rack, cluster, cc.OperatorImage)
+	desiredSts, err := resource.StatefulSetForRack(rack, cluster, cc.OperatorImage)
+	if err != nil {
+		return false, corev1.Container{}, err
+	}
+
 	desiredIdx, err := naming.FindSidecarInjectorContainer(desiredSts.Spec.Template.Spec.InitContainers)
 	if err != nil {
 		return false, corev1.Container{}, errors.Wrap(err, "find sidecar container in desired sts")

--- a/pkg/controllers/cluster/sync_test.go
+++ b/pkg/controllers/cluster/sync_test.go
@@ -23,7 +23,10 @@ func TestNextAction(t *testing.T) {
 
 	members := int32(3)
 	cluster := unit.NewSingleRackCluster(members)
-	rack := resource.StatefulSetForRack(cluster.Spec.Datacenter.Racks[0], cluster, operatorImage)
+	rack, err := resource.StatefulSetForRack(cluster.Spec.Datacenter.Racks[0], cluster, operatorImage)
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	clusterNewRackCreate := cluster.DeepCopy()
 	clusterNewRackCreate.Spec.Datacenter.Racks = append(

